### ISSUE TITLE
Sidebar: Allow resize

### ISSUE THREAD

**Issue by badosu** [extr0py/oni#1414]
Allow resizing of the sidebar grapping the separator.

**Comment by bryphe** [comment]
Makes sense, thanks for the suggestion @badosu . Once the API work in #1412 is in, this shouldn't be too difficult to integrate.

**Comment by badosu** [comment]
An update: this is how the Achievements pane is looking for me on the Sidebar. I am unable to resize it:

![screenshot from 2018-04-04 09-06-31](https://user-images.githubusercontent.com/347552/38306624-89d112e8-37e7-11e8-9747-71ffe20933ba.png)

**Comment by badosu** [comment]
Since https://github.com/onivim/oni/issues/2205 was merged, let's use this issue to track resizing the sidebar via grabbing the separator.
